V1

orthagonal opened the pull request.

orthagonal committed:
first commit, needs to be tidied before PR/code review

orthagonal committed:
working CLI version

-uses AWS profiles and env variables
-crops
-compresses
-uploads and prints s3 result

orthagonal committed:
lib + cli version

-lint not firing in my editor yet, will need lint fixes

orthagonal committed:
lint fixes

--added lintrc

jgallen23 commented on main.js:
```diff
@@ -0,0 +1,57 @@
+  const aws = awsAuth(argv);
+  // make sure working directory is present:
+  if (!fs.existsSync) {
+    fs.mkdirSync('build');
```
what is this for?

orthagonal replied:
Imagemin wants to read/write things from file, so I had it make a temp working directory.

jgallen23 commented on lib/commonAWS.js:
```diff
@@ -0,0 +1,62 @@
+'use strict';
```
think we could pull this out into a separate module?

orthagonal replied:
Definitely, we are just going to keep reusing this code for our AWS stuff.  Make a repo for it and add a readme with any specs you want and I'll switch over to it.

jgallen23 commented on cli.js:
```diff
@@ -0,0 +1,38 @@
+  },
+  bucket: {
+    describe: 's3 bucket to upload the image to',
+    default: 's3-bucket'
```
no default.  should be required to set

jgallen23 replied:
ping

orthagonal replied:
Committed the code for this yesterday, but was waiting on other feedback before pushing the commit (which causes Github to hide our comments pertaining to the old commit)

jgallen23 commented on lib/image.js:
```diff
@@ -0,0 +1,27 @@
+  const options = {
+    plugins: [imageminMozjpeg({ quality: level }), imageminPngquant({ quality: level })]
+  };
+  imagemin([file], 'build/', options).then(handleResult);
```
what about building to a temp directory?

orthagonal replied:
That's what the 'mkdir' is doing there, making sure the 'build' directory is present where the files are going to be stored.

jgallen23 reviewed (changes requested):
overall, looks good

jgallen23 commented on lib/image.js:
```diff
@@ -0,0 +1,27 @@
+  const options = {
+    plugins: [imageminMozjpeg({ quality: level }), imageminPngquant({ quality: level })]
+  };
+  imagemin([file], 'build/', options).then(handleResult);
```
that's not really a temp directory. look at os.tmpdir

orthagonal replied:
Oh I didn't know about that one, that's very nice.

orthagonal committed:
use os.tmpdir() instead of local 'build/' dir

will also now demand a --bucket option when run from cli

orthagonal committed:
uses npm's main and bin specifiers for lib/cli use

"s3put" should run the executable once installed
should load main.js when used as a module

jgallen23 commented:
Error: Cannot find module 'imagemin-mozjpeg' (I manually installed, but needs to be added to package.json)

jgallen23 commented:
```
20:51 $ node ~/libraries/s3put/cli.js --profile ft --bucket project-hub hero.jpg                                                                        
Unable to locate profile ft in file /home/ubuntu/.aws/config.
Falling back to us-east-1
{ ETag: '"d41d8cd98f00b204e9800998ecf8427e"',
  Location: 'https://project-hub.s3.amazonaws.com/hero.jpg',
  key: 'hero.jpg',
  Key: 'hero.jpg',
  Bucket: 'project-hub' }
```

uploaded a file with 0 bytes

orthagonal committed:
added and fixed unit tests

--updated to work correctly with file streams
--handles generating the tmp file and cleaning it up in a more accurate
fashion
--unit tests for each of the basic combinations of operations
--tests use env vars instead of hard-coding AWS vars
--added missing packages

orthagonal committed:
add --imagemagick flag
